plugins/baseplugin/src/main/java/org/apache/unomi/plugins/baseplugin/actions/IncrementPropertyAction.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.unomi.plugins.baseplugin.actions;
+
+import org.apache.commons.beanutils.PropertyUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.unomi.api.CustomItem;
+import org.apache.unomi.api.Event;
+import org.apache.unomi.api.Profile;
+import org.apache.unomi.api.actions.Action;
+import org.apache.unomi.api.actions.ActionExecutor;
+import org.apache.unomi.api.services.EventService;
+import org.apache.unomi.persistence.spi.PropertyHelper;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.lang.reflect.InvocationTargetException;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Map;
+
+public class IncrementPropertyAction implements ActionExecutor {
+    private static final Logger logger = LoggerFactory.getLogger(IncrementPropertyAction.class.getName());
+    private EventService eventService;
+
+    @Override
+    public int execute(final Action action, final Event event) {
+        boolean storeInSession = Boolean.TRUE.equals(action.getParameterValues().get("storeInSession"));
+        if (storeInSession && event.getSession() == null) {
+            return EventService.NO_CHANGE;
+        }
+
+        Profile profile = event.getProfile();
+        String propertyName = (String) action.getParameterValues().get("propertyName");
+        String propertyTarget = (String) action.getParameterValues().get("propertyTarget");
+        String rootPropertyName = propertyName.split("\\.")[0];
+        Object value = null;
+        Object propertyValue = 1;
+
+        try {
+            if (StringUtils.isNotEmpty(propertyTarget)) {
+                value = PropertyUtils.getNestedProperty(((CustomItem) event.getTarget()).getProperties(), propertyTarget);
+            }
+
+            if (value != null) {
+                if (value instanceof Integer) {
+                    if (profile.getProperty(rootPropertyName) != null) {
```

Comment:
What happen in case
propertyName = root.foo.bar
propertyTarget = bar
profile.root = {blabla:blabla}
event.bar = 15

the getNestedProperty will fail because the rootPropertyName will be "root" and the profile have the "root" property, but doesnt have the "root.foo".
To be verified, but I found the calculation and the usage of the "rootPropertyName" not clear, and hard to understand.

At the end it seem's only use to see if the profile have the property.